frontend/src/app/search/page.tsx

```diff
@@ -17,6 +20,12 @@ const searchFetcher = useMockData
 // }
 
 export default async function Search() {
+  const cookieStore = cookies();
+  const ffManager = new FeatureFlagsManager(cookieStore);
+  if (!ffManager.isFeatureEnabled("showSearchV0")) {
+    return notFound();
+  }
```

Comment:
We might be able to further clean this up and move it to the featureflag manager - but for now this is a good start